### DISCUSSION TITLE
Get Slot Content definations based on Campaign and Promotions

### DISCUSSION THREAD

**Discussion by ajaind1m** [SalesforceCommerceCloud/sfcc-ci#444] [Team Posts]
Hi All,

I am not able to find slot details from campaign/promotion manager API.

Is there any alternative to get details from campaign promotion not OCAPI?

**Comment by dgSFCC** [comment]
Hi ajaind1m,

Following line will get you all promotions configured in the BM:
var promotions = dw.campaign.PromotionMgr.getPromotions()

And from there if you write the following code, you will get the callout
messages for all these promotions:
var callOuts = [];
for each (var promo in promotions){
if(promo.calloutMsg){
callOuts.push(promo.calloutMsg);
}
}
return callOuts;

Here callOuts is an ArrayList with all the available callout messages.

Thanks and regards,
Dhruv Goel

**Comment by ajaind1m** [comment]
Hi Dhruv,

Thanks for reply, but here i am looking for solution to fetch slotID from campaign.
